src/pages/admin/donate/components/support-options/support-option-item/SupportOptionItem.tsx

```diff
@@ -0,0 +1,218 @@
+import { useEffect, useState } from 'react';
+import { Button } from '../../../../../../components/admin/button/Button';
+import { Input } from '../../input/Input';
+import './SupportOptionItem.scss';
+import { SupportOptionsType } from '../../../../../../types/admin/donate';
+import { ConfirmationModal } from '../../../../../../components/admin/confirmation-modal/ConfirmationModal';
+import { COMMON_TEXT_ADMIN } from '../../../../../../const/admin/common';
+import { DONATE_TEXT } from '../../../../../../const/admin/donate';
+import { SUPPORT_OPTIONS_VALIDATION_FUNCTIONS } from '../../../../../../validation/admin/bank-details-schema/bank-details-schema';
+
+export enum SupportOptionItemMode {
+    Create = 'create',
+    Edit = 'edit',
+    View = 'view',
+}
+
+interface ModalConfig {
+    title: string;
+    onConfirm: () => void;
+}
+
+export interface SupportOptionItemProps {
+    data?: SupportOptionsType;
+    initialMode?: SupportOptionItemMode;
+    onSave?: (name: string, value: string) => Promise<void>;
+    onCancel?: () => void;
+    onDelete?: () => Promise<void>;
+}
+
+export const SupportOptionItem = ({ data, initialMode, onSave, onCancel, onDelete }: SupportOptionItemProps) => {
+    const [mode, setMode] = useState<SupportOptionItemMode>(
+        initialMode ?? (data ? SupportOptionItemMode.View : SupportOptionItemMode.Create),
+    );
+    const [name, setName] = useState(data?.name ?? '');
+    const [value, setValue] = useState(data?.value ?? '');
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const [modalConfig, setModalConfig] = useState<ModalConfig | null>(null);
+    const [errors, setErrors] = useState<{ name?: string; value?: string }>({});
+
+    useEffect(() => {
+        setName(data?.name ?? '');
+        setValue(data?.value ?? '');
+        setErrors({});
+    }, [data]);
+
+    const hasEmptyFields = !name.trim() || !value.trim();
+
+    const hasChanges = () => {
+        return name !== (data?.name ?? '') || value !== (data?.value ?? '');
+    };
```

Comment:
Better to create this constants and use below
```suggestion
    const isViewMode = mode === SupportOptionItemMode.View;
    const isCreateMode = mode === SupportOptionItemMode.Create;
    const editable = !isViewMode;
    
    const hasChanges = name !== (data?.name ?? '') || value !== (data?.value ?? '');
    const hasErrors = !!errors.name || !!errors.value;
    const hasEmptyFields = !name.trim() || !value.trim();
    const isSaveDisabled = isSubmitting || hasEmptyFields || !hasChanges || hasErrors;
```